src/types.ts

```diff
@@ -41,39 +81,97 @@ export interface SuccessResponse {
   vm: DetectNote
 }
 
+/**
+ * Interface for error response of the bot detection API.
+ *
+ * @interface ErrorResponse
+ */
 export interface ErrorResponse {
   error: {
     code: ErrorCodes
     message: string
   }
 }
 
+/**
+ * Enum for bot detection error codes.
+ *
+ * @readonly
+ * @enum {string}
+ */
 export const enum ErrorCodes {
   BotdFailed = 'BotdFailed',
   DetectNotCalled = 'DetectNotCalled',
 }
 
+/**
+ * Interface for classes that represent a bot detector.
+ *
+ * @interface BotDetectorInterface
+ */
 export interface BotDetectorInterface {
+  /**
+   * Performs bot detection, internally it will make a network request to the server-side bot detection API.
+   *
+   * @param {(string|DetectOptions)} optionsOrTag Configuration options for bot detector. It's recommended to use the DetectOptions instead of the string tag.
+   * @returns {Promise<BotdResponse>} A promise to the instance of the bot detection response.
+   */
   detect(optionsOrTag?: string | DetectOptions): Promise<BotdResponse>
+
   /**
-   * @deprecated Will be removed in the next major version, use detect(options: DetectOptions) instead
+   * @deprecated Will be removed in the next major version, use detect(options: DetectOptions) instead.
    */
   detect(tag?: string): Promise<BotdResponse>
 
+  /**
+   * Collects all the components from the browser.
+   *
+   * @returns {Promise<ComponentDict>} A promise to the collected components.
+   */
   collect(): Promise<ComponentDict>
-  getResult(): Promise<BotdResponse>
 }
 
+/**
+ * Interface for configuration for the bot detector.
+ *
+ * @interface InitOptions
+ */
 export interface InitOptions {
+  /**
+   * A token required to access the server-side bot detection API.
+   */
   token: string
+
+  /**
+   * Represents mode for querying API. Default is 'requestId'.
+   */
   mode?: Modes
+
+  /**
+   * Optional endpoint for the server-side bot detection API.
+   *
+   * @todo Specify the endpoint for the server-side bot detection API once it's available.
```

Comment:
I added `@todo` because the option is available in the API, but there is no way to create an endpoint now. Once we have this feature, we can remove it and document what it is and where one can create it.